core/corehttp/gateway_handler.go

```diff
@@ -125,6 +125,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	i.addUserHeaders(w) // ok, _now_ write user's headers.
 	w.Header().Set("X-IPFS-Path", urlPath)
 
+	//robots-header to prevent search-engines from indexing the gateway because of the duplicate-content problematic when makeing the gateway public
+	w.Header().Set("X-Robots-Tag", "noindex")
```

Comment:
Correct me if I'm wrong, but I thought the goal for the native-browser implementation is that a browsers understand html-embed like `<img src="ifps://ipfs/[anyhash]" />` instead of binding the protocol to https://ipfs.io ( Like the firefox-ipfs-plugin does )

If we want to push a new protocol, it's valid thought. But if we want to redefine web-domains as something else than "identity of one or more IP addresses" it will be hard to distribute that new way of thinking.

I don't know how familiar you are with SEO but from my knowledge I would refrain from any possibility of creating duplicate content by publishing my content to more than one web-uri.

Your robots.txt from #1751 is the default robots which would also apply if no robots.txt is set.

I think a better way would be to create a sitemap.xml to display content to SE-crawlers. The ipfs-daemon should also parse the sitemap and add noindex to everything not included in there.